Weather/Weather/Adapter/WAdapter.m

```diff
@@ -0,0 +1,113 @@
+//
+
+#import "WAdapter.h"
+
+@implementation WAdapter
+
+NSString *const weatherList = @"list";
+NSString *const city = @"city";
+NSString *const cityName = @"name";
+NSString *const weatherAttribute = @"main";
+NSString *const weatherDescription = @"description";
+NSString *const maxTemperature = @"temp_max";
+NSString *const minTemperature = @"temp_min";
+NSString *const weather = @"weather";
+NSString *const weatherTemperature = @"temp";
+NSString *const dayName = @"dayName";
+
+
+- (NSDictionary *)rawDataToDict:(NSData *)data {
+    NSDictionary *newDict = [[NSDictionary alloc] init];
+    NSError* error;
+    newDict = [NSJSONSerialization JSONObjectWithData:data options:kNilOptions error:&error];
+    
+    return newDict;
+}
+- (NSMutableArray *)getFiveDays:(NSNumber *)length {
+    NSMutableArray *dayList = [[NSMutableArray alloc] init];
+    int i = 0;
+    while (i < [length integerValue]) {
+        NSCalendar *calendar = [NSCalendar currentCalendar];
+        NSDate *startDate = [calendar startOfDayForDate:[NSDate date]];
+        NSDate *sevenDaysFromNow = [calendar dateByAddingUnit:NSCalendarUnitDay value: i toDate: startDate options: NSCalendarMatchNextTime];
+        NSDateFormatter *dateFormatter = [[NSDateFormatter alloc] init];
+        [dateFormatter setDateFormat:@"EEEE"];
+        NSString *dayName = [dateFormatter stringFromDate:sevenDaysFromNow];
+        [dayList addObject:dayName];
+        ++i;
+    }
+    
+    return dayList;
+}
+
+- (NSMutableArray<NSDictionary *> *)dictToArray:(NSDictionary *)dict {
+    NSMutableArray *newArray = [[NSMutableArray alloc] init];
+    NSMutableArray *check = dict[weatherList];
+    NSDictionary *cityList = dict[city];
+    NSMutableArray *dayList = [self getFiveDays:[NSNumber numberWithInteger:[dict count]]];
+    NSString *location = cityList[cityName];
+    NSNumber *weatherListNum = [NSNumber numberWithInteger:[dayList count] - 1];
+    for (int i = 0; i <= [weatherListNum integerValue]; ++i) {
+        NSMutableArray *weatherArr = check[i][weather];
+        NSDictionary *main = check[i][weatherAttribute];
+        NSString *weather = [weatherArr objectAtIndex:0][weatherDescription];
+        NSNumber *highTemp = main[maxTemperature];
+        NSNumber *lowTemp = main[minTemperature];
+        NSNumber *current = main[weatherTemperature];
+        NSString *date = dayList[i];
+        NSDictionary *day = @{weatherDescription:weather, maxTemperature:highTemp, minTemperature:lowTemp, weatherTemperature:current, dayName:date, city:location};
+        [newArray addObject:day];
+    }
+    
+    return newArray;
+}
+
+
+- (WModel *)convertDataToModel:(NSData *)data {
+    NSDictionary *parsed = [self rawDataToDict:data];
+    NSMutableArray *dataList = [self dictToArray:parsed];
+    WModel *mInstance = [[WModel alloc] init];
+    [self updateModelwith:dataList of:mInstance];
+    
+    return mInstance;
+}
+
+
+- (NSMutableDictionary *)dataForCells:(WModel *)model {
```

Comment:
This doesn't seem like a method to be on the adapter. The adapter should only convert the server response into  model and data used later on. The Center seems more appropriate